### PR TITLE
Add strain exercise

### PR DESCRIPTION
**Pull request by BNAndras**
Related to #236

### CODE REVIEW

File: exercises/practice/strain/strain.coffee [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Strain
+  keep: (lst, predicate) -> 
+
+  discard: (lst, predicate) ->
```
Comment (ErikSchierboom) [inline exercises/practice/strain/strain.coffee:4]: Could these be class methods?

```suggestion
  @keep: (lst, predicate) -> 

  @discard: (lst, predicate) ->
```

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/practice/strain/strain.coffee
Comment (ladokp) [inline exercises/practice/strain/strain.coffee:1]: It looks like the tests are not synchronized with this change.

Comment (BNAndras) [reply exercises/practice/strain/strain.coffee:1]: Can you clarify please? I don't see any unsynced tests.
```
./bin/configlet sync --tests -e strain
Updating cached 'problem-specifications' data...
Checking exercises...
The `strain` exercise has up-to-date tests!
```

Comment (ladokp) [reply exercises/practice/strain/strain.coffee:1]: The initial code:
```
class Strain
  @keep: (lst, predicate) -> 

  @discard: (lst, predicate) ->

module.exports = Strain
```

produces the following error:
`TypeError: strain.keep is not a function`

Removing the @ makes the tests pass (when solution is implemented).

**Pull request merged**